src/linker/Linker.Dataflow/DynamicallyAccessedMembersTypeHierarchy.cs

```diff
@@ -142,6 +141,8 @@ public DynamicallyAccessedMembersTypeHierarchy (LinkContext context, MarkStep ma
 				return annotation;
 
 			// Apply the effective annotation for the type
+			using var _ = _scopeStack.PushScope (new MessageOrigin (type));
+			var reflectionPatternContext = new ReflectionPatternContext (_context, true, _scopeStack.CurrentScope.Origin, type);
```

Comment:
The `true` here has a substantial effect on the RUC and DAM interactions. At the very least this should have a comment on why. It took me a long time to decide what behavior I like, so I posted it as a comment on the main review thread (it's long).

Reply:
The true isn't really new - we have been reporting these warnings - but the origin is different than before. I added a comment here, and in MarkStep where the interesting decisions about RUC are made.